src/components/Button/Button.tsx

```diff
@@ -31,13 +35,29 @@ const buttonVariants = cva(
   },
 );
 
-export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
-  VariantProps<typeof buttonVariants> & {
-    asChild?: boolean;
-  };
+export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  asChild?: boolean;
+  size?: 'default' | 'sm' | 'lg' | 'icon';
```

Comment:
Gerade werden diese Typen doppelt definiert. 
Einmal hier und einmal implizit in Zeile 12 ff. Das gleiche gilt für die Varianten.
Hier würde ich Vorschlagen einmal die Typen zu definieren:
```typescript
type variants = 'default' | 'secondary' | 'destructive' | 'outline' | 'ghost' | 'link',
type sizes = 'default' | 'sm' | 'lg' | 'icon'
``` 

Anschließend einen Typ für die buttonVariants defienieren und dem cva mitgeben:
```typescript
type buttonVariants = {
    variant: Record<
      variants,
      string
    >;
    size: Record<sizes,string>;
  };
};

const buttonVariants = cva<buttonVariants>(
```

Und die Typen ebenfalls noch in den ButtonProps nutzen:
```typescript
export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  asChild?: boolean;
  size?: sizes
  variant?: variants
};
```

Dadurch können wir nicht an einer Stelle vergessen die Props zu erweitern sollten wir später eine Größe oder eine Variante hinzufügen.